go.mod

```diff
@@ -39,7 +39,7 @@ require (
 	golang.org/x/sync v0.12.0
 	golang.org/x/time v0.5.0
 	google.golang.org/grpc v1.63.2
-	google.golang.org/protobuf v1.33.0
+	google.golang.org/protobuf v1.36.5
```

Comment:
Ideally this change will not be part of this PR.  Instead, https://github.com/elastic/fleet-server/pull/4702 would be merged into `9.0` first, then this PR would be rebased on `9.0` and merged.